### PR TITLE
Require === and !==

### PR DESCRIPTION
**Pull request by koops76**
ping @Hainish @cowlicks.

See https://eslint.org/docs/rules/eqeqeq.

This change requires testing.

### CODE REVIEW

**Comment by cschanaj** [comment]
This changes might have some issues when handling `undefined` which required to be fixed, in particular

```bash
$ nodejs
> undefined !== null;
true
> undefined != null;
false
```

**Comment by koops76** [comment]
@cschanaj Can you find any?

**Comment by cschanaj** [comment]
AFAIK, the parsing of JSON use 

```
if (variable != null)
```

 which can be trivially rewrote by 

```
if (variable)
``` 

for the same functionalities. I am not sure about the rest of the code.

**Comment by koops76** [comment]
@Hainish @cowlicks Request review.

File: chromium/rules.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ RuleSet.prototype = {
   apply: function(urispec) {
     var returl = null;
     // If we're covered by an exclusion, go home
-    if (this.exclusions !== null) {
+    if (this.exclusions) {
```
Comment (cowlicks) [inline chromium/rules.js]: Why was this changed?

File: chromium/rules.js [hunk 2]
```diff
@@ -250,20 +250,20 @@ RuleSets.prototype = {
 
     // Read user prefs
     if (rule_set.name in this.ruleActiveStates) {
-      rule_set.active = (this.ruleActiveStates[rule_set.name] == "true");
+      rule_set.active = this.ruleActiveStates[rule_set.name];
```
Comment (cowlicks) [inline chromium/rules.js]: Does `rule_set.active`  go from being a bool, to a string?

File: chromium/rules.js [hunk 3]
```diff
@@ -250,20 +250,20 @@ RuleSets.prototype = {
...
     }
 
     var rules = ruletag["rule"];
     for (let rule of rules) {
-      if (rule["from"] != null && rule["to"] != null) {
+      if (rule["from"] && rule["to"]) {
```
Comment (cowlicks) [inline chromium/rules.js]: This code block will not execute in cases when it did before in the case where `rule["from"]` and `rule["to"]` are falsey, but not undefined. e.g:
```javascript
0 != null && false != null  // true
0 && false  // false
```

File: chromium/rules.js [hunk 4]
```diff
@@ -250,20 +250,20 @@ RuleSets.prototype = {
...
         rule_set.rules.push(new Rule(rule["from"], rule["to"]));
       }
     }
 
     var exclusions = ruletag["exclusion"];
-    if (exclusions != null) {
+    if (exclusions) {
       for (let exclusion of exclusions) {
-        if (exclusion != null) {
+        if (exclusion) {
```
Comment (cowlicks) [inline chromium/rules.js]: likewise, if `exclusion  = false`.
```javascript
exclusion != null // true
exclusion  // false
```

File: chromium/rules.js [hunk 5]
```diff
@@ -273,9 +273,9 @@ RuleSets.prototype = {
     }
 
     var cookierules = ruletag["securecookie"];
-    if (cookierules != null) {
+    if (cookierules) {
```
Comment (cowlicks) [inline chromium/rules.js]: In these cases, can you please explain why the variable will never be falsey but not `undefined` or `null`?

File: chromium/rules.js [hunk 6]
```diff
@@ -312,7 +312,7 @@ RuleSets.prototype = {
     // TODO: maybe promote this rule?
     this.targets.get(params.host).push(new_rule_set);
     if (new_rule_set.name in this.ruleActiveStates) {
-      new_rule_set.active = (this.ruleActiveStates[new_rule_set.name] == "true");
+      new_rule_set.active = this.ruleActiveStates[new_rule_set.name];
```
Comment (cowlicks) [inline chromium/rules.js]: This looks like another case we might have changed from bool to string.

File: chromium/rules.js [hunk 7]
```diff
@@ -359,7 +359,7 @@ RuleSets.prototype = {
     var platform = ruletag.getAttribute("platform");
     if (platform) {
       default_state = false;
-      if (platform == "mixedcontent" && settings.enableMixedRulesets) {
+      if (platform === "mixedcontent" && settings.enableMixedRulesets) {
```
Comment (cowlicks) [inline chromium/rules.js]: I'm guessing that `==` and `===` have the same operator precedence?  Could you add parentheses here to make it more explicit anyway

File: chromium/rules.js [hunk 8]
```diff
@@ -493,7 +493,7 @@ RuleSets.prototype = {
 
     var potentiallyApplicable = this.potentiallyApplicableRulesets(hostname);
     for (let ruleset of potentiallyApplicable) {
-      if (ruleset.cookierules !== null && ruleset.active) {
+      if (ruleset.cookierules && ruleset.active) {
```
Comment (cowlicks) [inline chromium/rules.js]: what if `ruleset.cookierules` is falsey but alse `!== null`?

File: chromium/rules.js (hunk 2 above)
Comment (koops76) [reply chromium/rules.js]: No, simply bad code.

File: chromium/rules.js (hunk 3 above)
Comment (koops76) [reply chromium/rules.js]: Thanks!

Comment (koops76) [reply chromium/rules.js]: Why would `from` and `to` be falsey? They should not be empty.

File: chromium/rules.js (hunk 4 above)
Comment (koops76) [reply chromium/rules.js]: Can `exclusion` be false?

File: chromium/rules.js (hunk 5 above)
Comment (koops76) [reply chromium/rules.js]: It will always be either an array (always truthy), or `null` or `undefined`.

File: chromium/rules.js (hunk 7 above)
Comment (koops76) [reply chromium/rules.js]: They do.

File: chromium/rules.js (hunk 8 above)
Comment (koops76) [reply chromium/rules.js]: It can only be an array or `null` or `undefined`.

File: chromium/rules.js (hunk 4 above)
Comment (cowlicks) [reply chromium/rules.js]: I don't know, I'm asking you. What can `exclusion` be? Is it possible for it to be falsey, but not null?

File: chromium/rules.js (hunk 1 above)
Comment (cowlicks) [reply chromium/rules.js]: Well, is that a bug? What exactly do we want to catch? `null`, `undefined`, anything else?

Comment (koops76) [reply chromium/rules.js]: Any falsey or non-array value I guess.

File: chromium/rules.js (hunk 4 above)
Comment (koops76) [reply chromium/rules.js]: I think it may be a string. In this case empty string is also falsey, but exclusions shouldn't be empty strings.

**Comment by koops76** [comment]
ping @Hainish @cowlicks.
